manager/flags.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2015 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package manager
+
+import (
+	"bytes"
+	"fmt"
+	"net/url"
+	"os"
+	"strings"
+)
+
+type Uri struct {
```

Comment:
Hmmm I tend to like the alternative we mentioned better: 

```Go
type SinkConfig struct {
    GCM GCMConfig `json:"gcm,omitempty"`
    InfluxDB InfluxDBConfig `json:"gcm,omitempty"`
    ...
}
```

It is stronger typed, self-documented, and easier on a REST API. It totally sucks for a flag interface though. WDYT? I am not against having the flag interface be different from the REST API.